csrc/scheduler/mma_utils.cpp

```diff
@@ -703,11 +704,51 @@ std::unordered_set<IterDomain*> getMmaDomainSet(
   return {mma_domains.begin(), mma_domains.end()};
 }
 
+bool isConsumerAllocationInnerIDProducerAllocationInnerID(
+    TensorView* consumer) {
+  // We are only looking at the allocation domains accross
+  // the producer and consumer of a LoadStoreOp
+  auto LdStOp = consumer->definition();
+  NVF_CHECK(dynamic_cast<LoadStoreOp*>(LdStOp));
+
+  // Get all the IDs from the innermost ID of the allocation domain of
+  // the consumer to the root domain of the consumer.
+  auto vals = DependencyCheck::getAllValsBetween(
+      {consumer->getRootDomain().begin(), consumer->getRootDomain().end()},
+      {consumer->getMaybeAllocationDomain().back()});
```

Comment:
Why is this needed. Can you not just use `idInConsumerRoot = consumer->getMaybeAllocationDomain().back()`?

Reply:
In the case of the matmul scheduler and the cache read TensorViews such as `bcr`/`acr`, the allocation domains are complicated (based on the leaf domains). The innermost ID from the allocation domain here is something that was generated by scheduling and ID (`k`). I traverse back to that ID here and then map it back to the producer.
 Let me know if you think this isn't correct.

Reply:
I see. I want to understand this a bit better.

This is called from here https://github.com/NVIDIA/Fuser/blob/5e0c89b289edcf52c4162713ee498ea86ce80564/csrc/scheduler/matmul.cpp#L1083-L1086
so the allocation domain of `acr` is a split/reorder of its logical domain in this case. So we might have something like the following:
```c++
acr = acw_smem->cacheAfter();
// acw_smem [ alloc: iS0{K} iS1{M} ]
//   iS2{Ko}, iS3{Ki} = split(iS0{K}, Ki)
//   iS4{Mo}, iS5{Mi} = split(iS1{M}, Mi}
// acr [ alloc: iS4{Mo} iS2{Ko} iS5{Mi} iS3{Ki} ]
```
If we supported multiple M dimensions then we would likely merge before this stage, so we would have
```c++
acr = acw_smem->cacheAfter();
// acw_smem [ alloc: iS0{M1} iS1{K} iS2{M2} ]
//   iS3{Ko}, iS4{Ki} = split(iS1{K}, Ki)
//   iS5{M1 * M2} = merge(iS0{M1}, iS2{M2})
//   iS6{Mo}, iS7{Mi} = split(iS5{M1 * M2}, Mi}
// acr [ alloc: iS6{Mo} iS3{Ko} iS7{Mi} iS4{Ki} ]
```
In this case it matters that the merge is in this order. If we did `iS5 = merge(iS2, iS0)` instead then I think we would not consider the allocation domain to be transposed? So should we actually traverse the IterDomain expressions from producer allocation to consumer allocation to determine whether the consumer's allocation domains are the innermost splits of (a possible inner merge of) the innermost producer allocation dimension?

Would it be simpler to just record the loop domain at the call site then actually set it after calling `ldMatrix`?

Reply:
@jacobhinkle, I'd like to better understand your comments. 
I created a PR (https://github.com/NVIDIA/Fuser/pull/2315/) to remove LdMatrixTranspose. 
Can we move this conversation to that PR?